subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -274,7 +275,10 @@ private boolean enableIncrementalCompilation(GroovyJavaJointCompileSpec spec) {
         if (getOptions().isIncremental() && spec.getSourceRoots().isEmpty()) {
             DeprecationLogger.nagUserOfDeprecatedBehaviour("Unable to infer source roots. Incremental Groovy compilation requires the source roots and has been disabled. Change the configuration of your sources or disable incremental Groovy compilation.");
         }
-        return getOptions().isIncremental() && !spec.getSourceRoots().isEmpty();
+        if (getOptions().isIncremental() && annotationProcessingConfigured(spec)) {
+            DeprecationLogger.nagUserOfDeprecatedBehaviour("Annotation processing is not supported when incremental Groovy compilation is enabled.");
```

Comment:
Should we make the deprecation warning more actionable? What we want to say is two things:
- We now disable incremental compilation since annotation processing is configured
- Having incremental compilation and annotation processing configured is deprecated and will be an error in Gradle 6.0

The options to fix this message are:
- Disable incremental compilation for GroovyCompile
- Remove the annotation processor configuration or use `-proc:none`

I wonder how much should go here and how much should go in the upgrade guide (`upgrading_version_5.adoc`).

Reply:
How about:
```
Incremental Groovy compilation has been disabled since Java annotation processors are configured.
Enabling incremental compilation and configuring Java annotation processors for Groovy compilation has been deprecated and will be removed in Gradle 6.0.
Disable incremental Groovy compilation or remove the Java annotation processor configuration.
```

So we need to use
```
DeprecationLogger.nagUserOfDeprecated("Incremental Groovy compilation has been disabled since Java annotation processors are configured. Enabling incremental compilation and configuring Java annotation processors for Groovy compilation",
"Disable incremental Groovy compilation or remove the Java annotation processor configuration.");
```

Reply:
Done, PTAL!